src/mono/mono/mini/interp/interp.c

```diff
@@ -1710,6 +1710,16 @@ ves_pinvoke_method (
 	interp_pop_lmf (&ext);
 
 #ifdef MONO_ARCH_HAVE_INTERP_PINVOKE_TRAMP
+	if (sig->call_convention == MONO_CALL_SWIFTCALL) {
+		int arg_index;
+		gpointer data = mono_arch_get_swift_error (&ccontext, sig, call_info, &arg_index);
+
+		// Perform an indirect store at arg_index stack location
+		if (data != NULL) {
+			stackval *result = (stackval*) STACK_ADD_BYTES (frame.stack, get_arg_offset (frame.imethod, sig, arg_index));
+			*(gpointer*)result->data.p = *(gpointer*)data;
+		}
```

Comment:
this could be under `#ifdef MONO_ARCH_HAVE_SWIFTCALL`.  Also in the case where we don't have swift interop support, we should throw `NotImplementedException` or something if we see a call with a swift calling convention

Reply:
Yes, we should also extend the tests.